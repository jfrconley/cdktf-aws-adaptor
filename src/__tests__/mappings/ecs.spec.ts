import { EcsCluster } from "@cdktf/provider-aws/lib/ecs-cluster/index.js";
import { EcsService } from "@cdktf/provider-aws/lib/ecs-service/index.js";
import { EcsTaskDefinition } from "@cdktf/provider-aws/lib/ecs-task-definition/index.js";
import { CfnCluster, CfnService, CfnTaskDefinition } from "aws-cdk-lib/aws-ecs";
import { describe } from "vitest";
import { itShouldMapCfnElementToTerraformResource } from "../helpers.js";

describe("ECS mappings", () => {
    itShouldMapCfnElementToTerraformResource(
        CfnCluster,
        {
            clusterName: "test-cluster",
            tags: [{
                key: "test-tag-key",
                value: "test-tag-value",
            }],
            configuration: {
                executeCommandConfiguration: {
                    logging: "DEFAULT",
                    kmsKeyId: "test-kms-key-id",
                    logConfiguration: {
                        cloudWatchLogGroupName: "test-log-group",
                        cloudWatchEncryptionEnabled: true,
                        s3BucketName: "test-bucket",
                        s3EncryptionEnabled: true,
                        s3KeyPrefix: "test-prefix",
                    },
                },
                managedStorageConfiguration: {
                    fargateEphemeralStorageKmsKeyId: "test-fargate-kms-key-id",
                    kmsKeyId: "test-kms-key-id",
                },
            },
            serviceConnectDefaults: {
                namespace: "test-namespace",
            },
            clusterSettings: [{
                name: "containerInsights",
                value: "enabled",
            }],
            capacityProviders: ["FARGATE", "FARGATE_SPOT"],
            defaultCapacityProviderStrategy: [{
                capacityProvider: "FARGATE",
                weight: 1,
                base: 1,
            }],
        },
        EcsCluster,
        {
            name: "test-cluster",
            tags: {
                "test-tag-key": "test-tag-value",
            },
            configuration: {
                executeCommandConfiguration: {
                    logging: "DEFAULT",
                    kmsKeyId: "test-kms-key-id",
                    logConfiguration: {
                        cloudWatchLogGroupName: "test-log-group",
                        cloudWatchEncryptionEnabled: true,
                        s3BucketName: "test-bucket",
                        s3BucketEncryptionEnabled: true,
                        s3KeyPrefix: "test-prefix",
                    },
                },
                managedStorageConfiguration: {
                    fargateEphemeralStorageKmsKeyId: "test-fargate-kms-key-id",
                    kmsKeyId: "test-kms-key-id",
                },
            },
            serviceConnectDefaults: {
                namespace: "test-namespace",
            },
            setting: [{
                name: "containerInsights",
                value: "enabled",
            }],
        },
    );

    itShouldMapCfnElementToTerraformResource(
        CfnService,
        {
            serviceName: "test-service",
            cluster: "test-cluster",
            taskDefinition: "test-task-definition",
            desiredCount: 2,
            availabilityZoneRebalancing: "ENABLED",
            enableExecuteCommand: true,
            enableEcsManagedTags: true,
            capacityProviderStrategy: [{
                capacityProvider: "test-provider",
                base: 1,
                weight: 100,
            }],
            deploymentConfiguration: {
                deploymentCircuitBreaker: {
                    enable: true,
                    rollback: true,
                },
                alarms: {
                    alarmNames: ["test-alarm"],
                    enable: true,
                    rollback: true,
                },
                maximumPercent: 200,
                minimumHealthyPercent: 50,
            },
            deploymentController: {
                type: "ECS",
            },
            healthCheckGracePeriodSeconds: 60,
            loadBalancers: [{
                containerName: "test-container",
                containerPort: 80,
                loadBalancerName: "test-lb",
                targetGroupArn: "test-target-group",
            }],
            networkConfiguration: {
                awsvpcConfiguration: {
                    subnets: ["subnet-1", "subnet-2"],
                    securityGroups: ["sg-1", "sg-2"],
                    assignPublicIp: "ENABLED",
                },
            },
            placementConstraints: [{
                type: "memberOf",
                expression: "attribute:ecs.instance-type =~ t3.*",
            }],
            schedulingStrategy: "REPLICA",
            serviceRegistries: [{
                registryArn: "test-registry",
                containerName: "test-container",
                containerPort: 80,
                port: 80,
            }],
            launchType: "FARGATE",
            role: "test-role",
            platformVersion: "LATEST",
            serviceConnectConfiguration: {
                enabled: true,
                namespace: "test-namespace",
                logConfiguration: {
                    logDriver: "awslogs",
                    options: {
                        "awslogs-group": "test-group",
                    },
                    secretOptions: [{
                        name: "test-secret",
                        valueFrom: "test-secret-arn",
                    }],
                },
                services: [{
                    portName: "test-port",
                    clientAliases: [{
                        port: 80,
                        dnsName: "test-dns",
                    }],
                    discoveryName: "test-discovery",
                    ingressPortOverride: 8080,
                    timeout: {
                        idleTimeoutSeconds: 60,
                        perRequestTimeoutSeconds: 60,
                    },
                    tls: {
                        issuerCertificateAuthority: {
                            awsPcaAuthorityArn: "test-aws-pca-authority-arn",
                        },
                        kmsKey: "test-kms-key",
                        roleArn: "test-role-arn",
                    },
                }],
            },
            tags: [{
                key: "test-tag-key",
                value: "test-tag-value",
            }],
            placementStrategies: [{
                type: "spread",
                field: "attribute:ecs.availability-zone",
            }],
            volumeConfigurations: [{
                name: "test-volume",
                managedEbsVolume: {
                    volumeInitializationRate: 100,
                    roleArn: "test-role-arn",
                    volumeType: "gp3",
                    iops: 3000,
                    encrypted: true,
                    filesystemType: "ext4",
                    kmsKeyId: "test-kms-key-id",
                    sizeInGiB: 20,
                    throughput: 125,
                    snapshotId: "test-snapshot",
                    tagSpecifications: [{
                        resourceType: "volume",
                        tags: [{
                            key: "test-volume-tag-key",
                            value: "test-volume-tag-value",
                        }],
                        propagateTags: "SERVICE",
                    }],
                },
            }],
            propagateTags: "SERVICE",
            vpcLatticeConfigurations: [{
                portName: "test-port",
                roleArn: "test-role-arn",
                targetGroupArn: "test-target-group-arn",
            }],
        },
        EcsService,
        {
            name: "test-service",
            cluster: "test-cluster",
            taskDefinition: "${data.aws_ecs_task_definition.resource_resource-task-definition_FCE63DB8.arn}",
            desiredCount: 2,
            availabilityZoneRebalancing: "ENABLED",
            enableExecuteCommand: true,
            enableEcsManagedTags: true,
            capacityProviderStrategy: [{
                capacityProvider: "test-provider",
                base: 1,
                weight: 100,
            }],
            deploymentCircuitBreaker: {
                enable: true,
                rollback: true,
            },
            alarms: {
                alarmNames: ["test-alarm"],
                enable: true,
                rollback: true,
            },
            deploymentMaximumPercent: 200,
            deploymentMinimumHealthyPercent: 50,
            deploymentController: {
                type: "ECS",
            },
            healthCheckGracePeriodSeconds: 60,
            loadBalancer: [{
                containerName: "test-container",
                containerPort: 80,
                targetGroupArn: "test-target-group",
                elbName: "test-lb",
            }],
            networkConfiguration: {
                subnets: ["subnet-1", "subnet-2"],
                securityGroups: ["sg-1", "sg-2"],
                assignPublicIp: true,
            },
            placementConstraints: [{
                type: "memberOf",
                expression: "attribute:ecs.instance-type =~ t3.*",
            }],
            schedulingStrategy: "REPLICA",
            serviceRegistries: {
                registryArn: "test-registry",
                containerName: "test-container",
                containerPort: 80,
                port: 80,
            },
            launchType: "FARGATE",
            iamRole: "test-role",
            platformVersion: "LATEST",
            serviceConnectConfiguration: {
                enabled: true,
                namespace: "test-namespace",
                logConfiguration: {
                    logDriver: "awslogs",
                    options: {
                        "awslogs-group": "test-group",
                    },
                    secretOption: [{
                        name: "test-secret",
                        valueFrom: "test-secret-arn",
                    }],
                },
                service: [{
                    tls: {
                        issuerCertAuthority: {
                            awsPcaAuthorityArn: "test-aws-pca-authority-arn",
                        },
                        kmsKey: "test-kms-key",
                        roleArn: "test-role-arn",
                    },
                    portName: "test-port",
                    discoveryName: "test-discovery",
                    clientAlias: {
                        port: 80,
                        dnsName: "test-dns",
                    },
                    ingressPortOverride: 8080,
                    timeout: {
                        idleTimeoutSeconds: 60,
                        perRequestTimeoutSeconds: 60,
                    },
                }],
            },
            tags: {
                "test-tag-key": "test-tag-value",
            },
            orderedPlacementStrategy: [{
                type: "spread",
                field: "attribute:ecs.availability-zone",
            }],
            propagateTags: "SERVICE",
            forceDelete: undefined,
            forceNewDeployment: undefined,
            waitForSteadyState: true,
            timeouts: undefined,
            triggers: undefined,
            volumeConfiguration: {
                name: "test-volume",
                managedEbsVolume: {
                    roleArn: "test-role-arn",
                    volumeType: "gp3",
                    iops: 3000,
                    encrypted: true,
                    fileSystemType: "ext4",
                    kmsKeyId: "test-kms-key-id",
                    sizeInGb: 20,
                    throughput: 125,
                    snapshotId: "test-snapshot",
                    tagSpecifications: [{
                        resourceType: "volume",
                        tags: {
                            "test-volume-tag-key": "test-volume-tag-value",
                        },
                        propagateTags: "SERVICE",
                    }],
                },
            },
            vpcLatticeConfigurations: [{
                portName: "test-port",
                roleArn: "test-role-arn",
                targetGroupArn: "test-target-group-arn",
            }],
        },
        ["volumeConfigurations.*.managedEbsVolume.volumeInitializationRate"],
    );

    itShouldMapCfnElementToTerraformResource(
        CfnTaskDefinition,
        {
            family: "test-task-definition",
            cpu: "256",
            inferenceAccelerators: [{
                deviceName: "test-device-name",
                deviceType: "test-device-type",
            }],
            ipcMode: "host",
            pidMode: "host",
            memory: "512",
            executionRoleArn: "arn:aws:iam::123456789012:role/ecsTaskExecutionRole",
            taskRoleArn: "arn:aws:iam::123456789012:role/ecsTaskRole",
            networkMode: "awsvpc",
            requiresCompatibilities: ["FARGATE"],
            containerDefinitions: [{
                name: "web",
                image: "nginx:latest",
                cpu: 256,
                memory: 512,
                memoryReservation: 256,
                restartPolicy: {
                    enabled: true,
                    ignoredExitCodes: [4],
                    restartAttemptPeriod: 0,
                },
                versionConsistency: "DEFAULT",
                essential: true,
                command: [],
                credentialSpecs: [],
                dependsOn: [],
                disableNetworking: false,
                dnsSearchDomains: [],
                dnsServers: [],
                dockerLabels: {},
                dockerSecurityOptions: [],
                entryPoint: [],
                environmentFiles: [],
                extraHosts: [],
                firelensConfiguration: {
                    type: "fluentbit",
                    options: {},
                },
                healthCheck: {
                    command: [],
                    interval: 0,
                    retries: 0,
                    startPeriod: 0,
                    timeout: 0,
                },
                hostname: "",
                interactive: false,
                links: [],
                linuxParameters: {
                    capabilities: {
                        add: [],
                        drop: [],
                    },
                    devices: [],
                    initProcessEnabled: false,
                    maxSwap: 0,
                    sharedMemorySize: 0,
                    swappiness: 0,
                    tmpfs: [],
                },
                portMappings: [{
                    containerPort: 80,
                    protocol: "tcp",
                    hostPort: 80,
                    name: "web",
                    appProtocol: "http",
                    containerPortRange: "",
                }],
                environment: [{
                    name: "ENV",
                    value: "prod",
                }],
                mountPoints: [{
                    sourceVolume: "efs-volume",
                    containerPath: "/usr/share/nginx/html",
                    readOnly: true,
                }],
                logConfiguration: {
                    logDriver: "awslogs",
                    options: {
                        "awslogs-group": "test-log-group",
                        "awslogs-region": "us-east-1",
                        "awslogs-stream-prefix": "web",
                    },
                    secretOptions: [],
                },
                privileged: false,
                pseudoTerminal: false,
                readonlyRootFilesystem: false,
                repositoryCredentials: {
                    credentialsParameter: "",
                },
                resourceRequirements: [],
                secrets: [],
                startTimeout: 0,
                stopTimeout: 0,
                systemControls: [],
                ulimits: [],
                user: "",
                volumesFrom: [],
                workingDirectory: "",
            }],
            volumes: [{
                name: "efs-volume",
                configuredAtLaunch: false,
                host: {
                    sourcePath: "",
                },
                efsVolumeConfiguration: {
                    filesystemId: "fs-12345678",
                    rootDirectory: "/",
                    transitEncryption: "ENABLED",
                    transitEncryptionPort: 2049,
                    authorizationConfig: {
                        accessPointId: "fsap-12345678",
                        iam: "ENABLED",
                    },
                },
                dockerVolumeConfiguration: {
                    autoprovision: false,
                    driver: "",
                    driverOpts: {
                        "type": "nfs",
                        "device": ":/test",
                    },
                    labels: {
                        "environment": "test",
                    },
                    scope: "shared",
                },
                fSxWindowsFileServerVolumeConfiguration: {
                    authorizationConfig: {
                        credentialsParameter: "",
                        domain: "",
                    },
                    rootDirectory: "",
                    fileSystemId: "",
                },
            }, {
                name: "docker-volume",
                configuredAtLaunch: false,
                host: {
                    sourcePath: "",
                },
                efsVolumeConfiguration: {
                    filesystemId: "",
                    rootDirectory: "",
                    transitEncryption: "ENABLED",
                    transitEncryptionPort: 2049,
                    authorizationConfig: {
                        accessPointId: "",
                        iam: "ENABLED",
                    },
                },
                dockerVolumeConfiguration: {
                    scope: "shared",
                    autoprovision: true,
                    driver: "local",
                    driverOpts: {
                        "type": "nfs",
                        "device": ":/test",
                    },
                    labels: {
                        "environment": "test",
                    },
                },
                fSxWindowsFileServerVolumeConfiguration: {
                    authorizationConfig: {
                        credentialsParameter: "",
                        domain: "",
                    },
                    rootDirectory: "",
                    fileSystemId: "",
                },
            }],
            proxyConfiguration: {
                type: "APPMESH",
                containerName: "envoy",
                proxyConfigurationProperties: [{
                    name: "ProxyIngressPort",
                    value: "15000",
                }, {
                    name: "ProxyEgressPort",
                    value: "15001",
                }],
            },
            placementConstraints: [{
                type: "memberOf",
                expression: "attribute:ecs.instance-type =~ t3.*",
            }],
            enableFaultInjection: true,
            ephemeralStorage: {
                sizeInGiB: 50,
            },
            runtimePlatform: {
                cpuArchitecture: "X86_64",
                operatingSystemFamily: "LINUX",
            },
            tags: [{
                key: "Environment",
                value: "Production",
            }],
        },
        EcsTaskDefinition,
        {
            family: "test-task-definition",
            cpu: "256",
            memory: "512",
            executionRoleArn: "arn:aws:iam::123456789012:role/ecsTaskExecutionRole",
            taskRoleArn: "arn:aws:iam::123456789012:role/ecsTaskRole",
            networkMode: "awsvpc",
            requiresCompatibilities: ["FARGATE"],
            containerDefinitions:
                "${jsonencode([{\"Cpu\" = 256, \"DisableNetworking\" = false, \"Environment\" = [{\"Name\" = \"ENV\", \"Value\" = \"prod\"}], \"Essential\" = true, \"FirelensConfiguration\" = {\"Type\" = \"fluentbit\"}, \"HealthCheck\" = {\"Interval\" = 0, \"Retries\" = 0, \"StartPeriod\" = 0, \"Timeout\" = 0}, \"Hostname\" = \"\", \"Image\" = \"nginx:latest\", \"Interactive\" = false, \"LinuxParameters\" = {\"Capabilities\" = {}, \"InitProcessEnabled\" = false, \"MaxSwap\" = 0, \"SharedMemorySize\" = 0, \"Swappiness\" = 0}, \"LogConfiguration\" = {\"LogDriver\" = \"awslogs\", \"Options\" = {\"awslogs-group\" = \"test-log-group\", \"awslogs-region\" = \"us-east-1\", \"awslogs-stream-prefix\" = \"web\"}}, \"Memory\" = 512, \"MemoryReservation\" = 256, \"MountPoints\" = [{\"ContainerPath\" = \"/usr/share/nginx/html\", \"ReadOnly\" = true, \"SourceVolume\" = \"efs-volume\"}], \"Name\" = \"web\", \"PortMappings\" = [{\"AppProtocol\" = \"http\", \"ContainerPort\" = 80, \"ContainerPortRange\" = \"\", \"HostPort\" = 80, \"Name\" = \"web\", \"Protocol\" = \"tcp\"}], \"Privileged\" = false, \"PseudoTerminal\" = false, \"ReadonlyRootFilesystem\" = false, \"RepositoryCredentials\" = {\"CredentialsParameter\" = \"\"}, \"RestartPolicy\" = {\"Enabled\" = true, \"IgnoredExitCodes\" = [4], \"RestartAttemptPeriod\" = 0}, \"StartTimeout\" = 0, \"StopTimeout\" = 0, \"User\" = \"\", \"VersionConsistency\" = \"DEFAULT\", \"WorkingDirectory\" = \"\"}])}",
            volume: [{
                dockerVolumeConfiguration: {
                    scope: "shared",
                    autoprovision: false,
                    driver: "",
                    driverOpts: {
                        "type": "nfs",
                        "device": ":/test",
                    },
                    labels: {
                        "environment": "test",
                    },
                },
                fsxWindowsFileServerVolumeConfiguration: {
                    fileSystemId: "",
                    rootDirectory: "",
                    authorizationConfig: {
                        credentialsParameter: "",
                        domain: "",
                    },
                },
                configureAtLaunch: false,
                hostPath: "",
                name: "efs-volume",
                efsVolumeConfiguration: {
                    fileSystemId: "fs-12345678",
                    rootDirectory: "/",
                    transitEncryption: "ENABLED",
                    transitEncryptionPort: 2049,
                    authorizationConfig: {
                        accessPointId: "fsap-12345678",
                        iam: "ENABLED",
                    },
                },
            }, {
                name: "docker-volume",
                configureAtLaunch: false,
                hostPath: "",
                efsVolumeConfiguration: {
                    fileSystemId: "",
                    rootDirectory: "",
                    transitEncryption: "ENABLED",
                    transitEncryptionPort: 2049,
                    authorizationConfig: {
                        accessPointId: "",
                        iam: "ENABLED",
                    },
                },
                dockerVolumeConfiguration: {
                    scope: "shared",
                    autoprovision: true,
                    driver: "local",
                    driverOpts: {
                        "type": "nfs",
                        "device": ":/test",
                    },
                    labels: {
                        "environment": "test",
                    },
                },
                fsxWindowsFileServerVolumeConfiguration: {
                    fileSystemId: "",
                    rootDirectory: "",
                    authorizationConfig: {
                        credentialsParameter: "",
                        domain: "",
                    },
                },
            }],
            proxyConfiguration: {
                type: "APPMESH",
                containerName: "envoy",
                properties: {
                    "ProxyIngressPort": "15000",
                    "ProxyEgressPort": "15001",
                },
            },
            placementConstraints: [{
                type: "memberOf",
                expression: "attribute:ecs.instance-type =~ t3.*",
            }],
            enableFaultInjection: true,
            ephemeralStorage: {
                sizeInGib: 50,
            },
            ipcMode: "host",
            pidMode: "host",
            skipDestroy: undefined,
            trackLatest: undefined,
            inferenceAccelerator: undefined,
            runtimePlatform: {
                cpuArchitecture: "X86_64",
                operatingSystemFamily: "LINUX",
            },
            tags: {
                "Environment": "Production",
            },
        },
    );
});
