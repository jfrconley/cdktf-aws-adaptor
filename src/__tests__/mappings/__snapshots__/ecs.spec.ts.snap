// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ECS mappings > Should map AWS::ECS::Cluster 1`] = `
"{
  "provider": {
    "aws": [
      {
        "alias": "us_east_1",
        "region": "us-east-1"
      }
    ]
  },
  "resource": {
    "aws_ecs_cluster": {
      "resource_22C949BF": {
        "configuration": {
          "execute_command_configuration": {
            "kms_key_id": "test-kms-key-id",
            "log_configuration": {
              "cloud_watch_encryption_enabled": true,
              "cloud_watch_log_group_name": "test-log-group",
              "s3_bucket_encryption_enabled": true,
              "s3_bucket_name": "test-bucket",
              "s3_key_prefix": "test-prefix"
            },
            "logging": "DEFAULT"
          },
          "managed_storage_configuration": {
            "fargate_ephemeral_storage_kms_key_id": "test-fargate-kms-key-id",
            "kms_key_id": "test-kms-key-id"
          }
        },
        "name": "test-cluster",
        "service_connect_defaults": {
          "namespace": "test-namespace"
        },
        "setting": [
          {
            "name": "containerInsights",
            "value": "enabled"
          }
        ],
        "tags": {
          "test-tag-key": "test-tag-value"
        }
      }
    },
    "aws_ecs_cluster_capacity_providers": {
      "resource_resource-capacity-providers_00D4E0EA": {
        "capacity_providers": [
          "FARGATE",
          "FARGATE_SPOT"
        ],
        "cluster_name": "\${aws_ecs_cluster.resource_22C949BF.name}",
        "default_capacity_provider_strategy": [
          {
            "base": 1,
            "capacity_provider": "FARGATE",
            "weight": 1
          }
        ]
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "/terraform.test-stack.tfstate"
      }
    },
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "5.85.0"
      }
    }
  },
  "variable": {
    "resource_refs_CDA17697": {
      "default": "\${aws_ecs_cluster.resource_22C949BF.arn}",
      "type": "string"
    }
  }
}"
`;

exports[`ECS mappings > Should map AWS::ECS::Service 1`] = `
"{
  "data": {
    "aws_ecs_task_definition": {
      "resource_resource-task-definition_FCE63DB8": {
        "task_definition": "test-task-definition"
      }
    }
  },
  "provider": {
    "aws": [
      {
        "alias": "us_east_1",
        "region": "us-east-1"
      }
    ]
  },
  "resource": {
    "aws_ecs_service": {
      "resource_22C949BF": {
        "alarms": {
          "alarm_names": [
            "test-alarm"
          ],
          "enable": true,
          "rollback": true
        },
        "availability_zone_rebalancing": "ENABLED",
        "capacity_provider_strategy": [
          {
            "base": 1,
            "capacity_provider": "test-provider",
            "weight": 100
          }
        ],
        "cluster": "test-cluster",
        "deployment_circuit_breaker": {
          "enable": true,
          "rollback": true
        },
        "deployment_controller": {
          "type": "ECS"
        },
        "deployment_maximum_percent": 200,
        "deployment_minimum_healthy_percent": 50,
        "desired_count": 2,
        "enable_ecs_managed_tags": true,
        "enable_execute_command": true,
        "health_check_grace_period_seconds": 60,
        "iam_role": "test-role",
        "launch_type": "FARGATE",
        "load_balancer": [
          {
            "container_name": "test-container",
            "container_port": 80,
            "elb_name": "test-lb",
            "target_group_arn": "test-target-group"
          }
        ],
        "name": "test-service",
        "network_configuration": {
          "assign_public_ip": true,
          "security_groups": [
            "sg-1",
            "sg-2"
          ],
          "subnets": [
            "subnet-1",
            "subnet-2"
          ]
        },
        "ordered_placement_strategy": [
          {
            "field": "attribute:ecs.availability-zone",
            "type": "spread"
          }
        ],
        "placement_constraints": [
          {
            "expression": "attribute:ecs.instance-type =~ t3.*",
            "type": "memberOf"
          }
        ],
        "platform_version": "LATEST",
        "propagate_tags": "SERVICE",
        "scheduling_strategy": "REPLICA",
        "service_connect_configuration": {
          "enabled": true,
          "log_configuration": {
            "log_driver": "awslogs",
            "options": {
              "awslogs-group": "test-group"
            },
            "secret_option": [
              {
                "name": "test-secret",
                "value_from": "test-secret-arn"
              }
            ]
          },
          "namespace": "test-namespace",
          "service": [
            {
              "client_alias": {
                "dns_name": "test-dns",
                "port": 80
              },
              "discovery_name": "test-discovery",
              "ingress_port_override": 8080,
              "port_name": "test-port",
              "timeout": {
                "idle_timeout_seconds": 60,
                "per_request_timeout_seconds": 60
              },
              "tls": {
                "issuer_cert_authority": {
                  "aws_pca_authority_arn": "test-aws-pca-authority-arn"
                },
                "kms_key": "test-kms-key",
                "role_arn": "test-role-arn"
              }
            }
          ]
        },
        "service_registries": {
          "container_name": "test-container",
          "container_port": 80,
          "port": 80,
          "registry_arn": "test-registry"
        },
        "tags": {
          "test-tag-key": "test-tag-value"
        },
        "task_definition": "\${data.aws_ecs_task_definition.resource_resource-task-definition_FCE63DB8.arn}",
        "volume_configuration": {
          "managed_ebs_volume": {
            "encrypted": true,
            "file_system_type": "ext4",
            "iops": 3000,
            "kms_key_id": "test-kms-key-id",
            "role_arn": "test-role-arn",
            "size_in_gb": 20,
            "snapshot_id": "test-snapshot",
            "tag_specifications": [
              {
                "propagate_tags": "SERVICE",
                "resource_type": "volume",
                "tags": {
                  "test-volume-tag-key": "test-volume-tag-value"
                }
              }
            ],
            "throughput": 125,
            "volume_type": "gp3"
          },
          "name": "test-volume"
        },
        "vpc_lattice_configurations": [
          {
            "port_name": "test-port",
            "role_arn": "test-role-arn",
            "target_group_arn": "test-target-group-arn"
          }
        ],
        "wait_for_steady_state": true
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "/terraform.test-stack.tfstate"
      }
    },
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "5.85.0"
      }
    }
  },
  "variable": {
    "resource_refs_CDA17697": {
      "default": "\${join(\\"\\", [aws_ecs_service.resource_22C949BF.name, \\",\\", aws_ecs_service.resource_22C949BF.id])}",
      "type": "string"
    }
  }
}"
`;

exports[`ECS mappings > Should map AWS::ECS::TaskDefinition 1`] = `
"{
  "provider": {
    "aws": [
      {
        "alias": "us_east_1",
        "region": "us-east-1"
      }
    ]
  },
  "resource": {
    "aws_ecs_task_definition": {
      "resource_22C949BF": {
        "container_definitions": "\${jsonencode([{\\"Cpu\\" = 256, \\"DisableNetworking\\" = false, \\"Environment\\" = [{\\"Name\\" = \\"ENV\\", \\"Value\\" = \\"prod\\"}], \\"Essential\\" = true, \\"FirelensConfiguration\\" = {\\"Type\\" = \\"fluentbit\\"}, \\"HealthCheck\\" = {\\"Interval\\" = 0, \\"Retries\\" = 0, \\"StartPeriod\\" = 0, \\"Timeout\\" = 0}, \\"Hostname\\" = \\"\\", \\"Image\\" = \\"nginx:latest\\", \\"Interactive\\" = false, \\"LinuxParameters\\" = {\\"Capabilities\\" = {}, \\"InitProcessEnabled\\" = false, \\"MaxSwap\\" = 0, \\"SharedMemorySize\\" = 0, \\"Swappiness\\" = 0}, \\"LogConfiguration\\" = {\\"LogDriver\\" = \\"awslogs\\", \\"Options\\" = {\\"awslogs-group\\" = \\"test-log-group\\", \\"awslogs-region\\" = \\"us-east-1\\", \\"awslogs-stream-prefix\\" = \\"web\\"}}, \\"Memory\\" = 512, \\"MemoryReservation\\" = 256, \\"MountPoints\\" = [{\\"ContainerPath\\" = \\"/usr/share/nginx/html\\", \\"ReadOnly\\" = true, \\"SourceVolume\\" = \\"efs-volume\\"}], \\"Name\\" = \\"web\\", \\"PortMappings\\" = [{\\"AppProtocol\\" = \\"http\\", \\"ContainerPort\\" = 80, \\"ContainerPortRange\\" = \\"\\", \\"HostPort\\" = 80, \\"Name\\" = \\"web\\", \\"Protocol\\" = \\"tcp\\"}], \\"Privileged\\" = false, \\"PseudoTerminal\\" = false, \\"ReadonlyRootFilesystem\\" = false, \\"RepositoryCredentials\\" = {\\"CredentialsParameter\\" = \\"\\"}, \\"RestartPolicy\\" = {\\"Enabled\\" = true, \\"IgnoredExitCodes\\" = [4], \\"RestartAttemptPeriod\\" = 0}, \\"StartTimeout\\" = 0, \\"StopTimeout\\" = 0, \\"User\\" = \\"\\", \\"VersionConsistency\\" = \\"DEFAULT\\", \\"WorkingDirectory\\" = \\"\\"}])}",
        "cpu": "256",
        "enable_fault_injection": true,
        "ephemeral_storage": {
          "size_in_gib": 50
        },
        "execution_role_arn": "arn:aws:iam::123456789012:role/ecsTaskExecutionRole",
        "family": "test-task-definition",
        "ipc_mode": "host",
        "memory": "512",
        "network_mode": "awsvpc",
        "pid_mode": "host",
        "placement_constraints": [
          {
            "expression": "attribute:ecs.instance-type =~ t3.*",
            "type": "memberOf"
          }
        ],
        "proxy_configuration": {
          "container_name": "envoy",
          "properties": {
            "ProxyEgressPort": "15001",
            "ProxyIngressPort": "15000"
          },
          "type": "APPMESH"
        },
        "requires_compatibilities": [
          "FARGATE"
        ],
        "runtime_platform": {
          "cpu_architecture": "X86_64",
          "operating_system_family": "LINUX"
        },
        "tags": {
          "Environment": "Production"
        },
        "task_role_arn": "arn:aws:iam::123456789012:role/ecsTaskRole",
        "volume": [
          {
            "configure_at_launch": false,
            "docker_volume_configuration": {
              "autoprovision": false,
              "driver": "",
              "driver_opts": {
                "device": ":/test",
                "type": "nfs"
              },
              "labels": {
                "environment": "test"
              },
              "scope": "shared"
            },
            "efs_volume_configuration": {
              "authorization_config": {
                "access_point_id": "fsap-12345678",
                "iam": "ENABLED"
              },
              "file_system_id": "fs-12345678",
              "root_directory": "/",
              "transit_encryption": "ENABLED",
              "transit_encryption_port": 2049
            },
            "fsx_windows_file_server_volume_configuration": {
              "authorization_config": {
                "credentials_parameter": "",
                "domain": ""
              },
              "file_system_id": "",
              "root_directory": ""
            },
            "host_path": "",
            "name": "efs-volume"
          },
          {
            "configure_at_launch": false,
            "docker_volume_configuration": {
              "autoprovision": true,
              "driver": "local",
              "driver_opts": {
                "device": ":/test",
                "type": "nfs"
              },
              "labels": {
                "environment": "test"
              },
              "scope": "shared"
            },
            "efs_volume_configuration": {
              "authorization_config": {
                "access_point_id": "",
                "iam": "ENABLED"
              },
              "file_system_id": "",
              "root_directory": "",
              "transit_encryption": "ENABLED",
              "transit_encryption_port": 2049
            },
            "fsx_windows_file_server_volume_configuration": {
              "authorization_config": {
                "credentials_parameter": "",
                "domain": ""
              },
              "file_system_id": "",
              "root_directory": ""
            },
            "host_path": "",
            "name": "docker-volume"
          }
        ]
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "/terraform.test-stack.tfstate"
      }
    },
    "required_providers": {
      "aws": {
        "source": "aws",
        "version": "5.85.0"
      }
    }
  },
  "variable": {
    "resource_refs_CDA17697": {
      "default": "\${aws_ecs_task_definition.resource_22C949BF.arn}",
      "type": "string"
    }
  }
}"
`;
